dxcint/dxcint/RegisteredTest.py

```diff
@@ -0,0 +1,183 @@
+import os
+import shlex
+import random
+import dxpy
+import json
+
+import subprocess as sp
+from typing import List, Optional, Dict, Union
+
+from dxcint.utils import rm_prefix
+from dxcint.Messenger import Messenger
+from dxcint.Context import Context
+
+
+class RegisteredTestError(Exception):
+    """
+    Class to handle RegisteredTest errors
+    """
+
+
+class RegisteredTest(object):
+    def __init__(
+            self,
+            src_file: str,
+            category: str,
+            test_name: str,
+            context: Context
+    ):
+        self._context = context
+        self._src_file = src_file
+        self._category = category
+        self._test_name = test_name
+        self._language = rm_prefix(os.path.basename(src_file), test_name).strip(".")
+        self._messenger = None
+        self._exec_id = None
+        self._job_id = None
+        self._test_results = None
+        self._executable_type_switch = {
+            "workflow": dxpy.DXWorkflow,
```

Comment:
We also run GlobalWorkflow (e.g [here](https://github.com/dnanexus/dxCompiler/blob/develop/scripts/run_multiuser_tests.py#L218)), separately in the integration pipeline though https://github.com/dnanexus/dxCompiler/blob/develop/.github/workflows/integration.yml#L71 We don't compile global workflows (and apps) with dxCompiler.

Reply:
cool, sure. Will add that no problem - as soon as I start implementing other subclasses of `RegisteredTest`. Compiling or not - will be handled in those subclasses. I'll just override [this method](https://github.com/dnanexus/dxCompiler/blob/3edc6089761c2a54216160eeb67550fdc433820f/dxcint/dxcint/RegisteredTest.py#L99) to skip compilation